docs/testing/test-get-started-md.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+import os
+import platform
+import re
+import shlex
+import shutil
+import subprocess
+import sys
+import tempfile
+import difflib
+
+REQUIRED_BINARIES = ["git", "ssh-keygen"]
+SNIPPET_PATTERN = r"```bash\n([\s\S]*?)\n```"
+EXPECTED_OUTPUT_FILENAME = "tester-expected-unix.txt"
+GET_STARTED_FILENAME = "get-started.md"
+
+# Validate that we have all the binaries required to run the test commands
+def check_binaries():
+    for p in REQUIRED_BINARIES:
+        if not shutil.which(p):
+            raise Exception(f"required command {p} not found")
+
+
+def test_commands():
+    curr_path = os.getcwd()
+    docs_path = os.path.join(curr_path + "/docs")
+    testing_path = os.path.join(docs_path + "/testing")
+    keys_path = os.path.join(testing_path + "/keys")
+    os.chdir(testing_path)
+
+    # Prepare testing directory
+    match platform.system():
+        case "Linux":
+            expected_output_file = os.path.realpath(os.path.join(testing_path, EXPECTED_OUTPUT_FILENAME))
+        case "Darwin":
+            expected_output_file = os.path.realpath(os.path.join(testing_path, EXPECTED_OUTPUT_FILENAME))
+        case "Windows":
+            raise SystemExit("Windows is not supported at this time.")
+        case _:
+            raise SystemExit("Unknown platform.")
+    
+    get_started_file = os.path.realpath(os.path.join(docs_path, GET_STARTED_FILENAME))
+    tmp_dir = os.path.realpath(tempfile.mkdtemp())
+    os.chdir(tmp_dir)
+
+    # Copy keys used to make git hashes deterministic    
+    repo_path = os.path.join(tmp_dir + "/repo")
+    repo_keys_path = os.path.join(tmp_dir + "/keys")
+    shutil.copytree(keys_path, repo_keys_path)
+    os.chdir(repo_keys_path)
+    os.chmod("developer", 0o0600)
+    
+    try:
+        with open(expected_output_file) as fp1, open(get_started_file) as fp2:
+            expected_output = fp1.read()
+            get_started = fp2.read()
+            snippets = re.findall(SNIPPET_PATTERN, get_started)
+
+            for i, snippet in enumerate(snippets):
+                snippets[i] = snippet.replace("$ ", "")
+            script = "\nset -x\n{}".format("\n".join(snippets))
+            script += "\ngittuf verify-ref main" # Workaround for non-deterministic hashes
+
+            # Set some environment variables to make git hashes deterministic
+            cmd_env = os.environ.copy()
+
+            cmd_env["GIT_AUTHOR_NAME"] = "Example E"
+            cmd_env["GIT_AUTHOR_EMAIL"] = "example@example.org"
+            cmd_env["GIT_AUTHOR_DATE"] = "Mon, 3 Jun 2024 10:00:00 -0400"
+            cmd_env["GIT_COMMITTER_NAME"] = "Example E"
+            cmd_env["GIT_COMMITTER_EMAIL"] = "example@example.org"
```

Comment:
https://github.com/gittuf/gittuf/blob/main/internal/gitinterface/common.go#L17-L20 for consistency, let's use these values? :D